src/Voron/Impl/Journal/JournalFile.cs

```diff
@@ -150,19 +153,86 @@ public void Write(LowLevelTransaction tx, CompressedPagesResult pages)
 
         public void InitFrom(StorageEnvironment storageEnvironment, JournalReader journalReader, List<TransactionHeader> transactionHeaders)
         {
-            storageEnvironment.UpdateJournal(this, journalReader.Next4Kb);
+            storageEnvironment.UpdateJournal(Number, journalReader.Next4Kb);
             _transactionHeaders = [.. transactionHeaders];
         }
 
-        public bool DeleteOnClose
+        public bool ShouldDelete
         {
             set
             {
-                var writer = journalWriter;
+                var writer = _journalWriter;
 
                 if (writer != null)
-                    writer.DeleteOnClose = value;
+                    writer.ShouldDelete = value;
+            }
+        }
+
+        public bool HasLegacyTransaction
+        {
+            get
+            {
+                foreach (var tx in _transactionHeaders)
+                {
+                    // if this journal contains any transaction with empty database 
+                    // then we cannot use it for current writes, since it may be a 
+                    // root environment and confuse any branch env reading from it
+                    if (tx.JournalId == Guid.Empty)
+                        return true;
+                }
+
+                return false;
             }
         }
+
+        public void SetTransactionFrom(Pal.journal_entry journalEntry)
+        {
+            _transactionHeaders.Add(*(TransactionHeader*)journalEntry.Base);
+        }
+
+        /// <summary>
+        ///  A journal file is valid for a journal id if:
+        /// - There are no existing transactions in the journal for this journal id
+        /// - There *are* existing transactions in the journal *and* that journal is a hard link
+        ///
+        /// The issue is that we may have a snapshot / manual file move that would result in breaking
+        /// of the hard link between shared journals. Consider the case of an index & database that have
```

Comment:
1. Will it still be possible to get raw data of an index (and shared journals) and inspect it?
2. Will it be possible to get just a database data (and shared journals) and load it without indexes? Won't it fail on the recovery due to encountering transactions coming from indexes environments?